# HG changeset patch
# User Arlo Breault <arlolra@gmail.com>
# Date 1410462708 25200
#      Thu Sep 11 12:11:48 2014 -0700
# Node ID 170790c9ae5423c1c36118ab0d38203e2fb16c68
# Parent  ff0c1a9ea8eb1b65c56b9e46f7fd739e8916f7cb
Add PrepareForSending to purpleConversation

 * Needed for transform API.

Bug: 983347

diff --git a/purplexpcom/src/purpleConvChat.h b/purplexpcom/src/purpleConvChat.h
--- a/purplexpcom/src/purpleConvChat.h
+++ b/purplexpcom/src/purpleConvChat.h
@@ -55,16 +55,20 @@ public:
     return purpleConversation::Close();
   }
   NS_IMETHOD AddObserver(nsIObserver *aObserver) {
     return purpleConversation::AddObserver(aObserver);
   }
   NS_IMETHOD RemoveObserver(nsIObserver *aObserver) {
     return purpleConversation::RemoveObserver(aObserver);
   }
+  NS_IMETHOD PrepareForSending(imIOutgoingMessage *aMsg, uint32_t *aMsgCount,
+                               char16_t * **aMsgs) {
+    return purpleConversation::PrepareForSending(aMsg, aMsgCount, aMsgs);
+  }
 
   // Keep the SendMsg method here
   NS_IMETHOD SendMsg(const nsACString & aMsg);
 
   purpleConvChat();
 
 private:
   ~purpleConvChat();
diff --git a/purplexpcom/src/purpleConvIM.h b/purplexpcom/src/purpleConvIM.h
--- a/purplexpcom/src/purpleConvIM.h
+++ b/purplexpcom/src/purpleConvIM.h
@@ -50,16 +50,20 @@ public:
     return purpleConversation::Close();
   }
   NS_IMETHOD AddObserver(nsIObserver *aObserver) {
     return purpleConversation::AddObserver(aObserver);
   }
   NS_IMETHOD RemoveObserver(nsIObserver *aObserver) {
     return purpleConversation::RemoveObserver(aObserver);
   }
+  NS_IMETHOD PrepareForSending(imIOutgoingMessage *aMsg, uint32_t *aMsgCount,
+                               char16_t * **aMsgs) {
+    return purpleConversation::PrepareForSending(aMsg, aMsgCount, aMsgs);
+  }
 
   // Keep the SendMsg and SendTyping methods here.
   NS_IMETHOD SendMsg(const nsACString & aMsg);
   NS_IMETHODIMP SendTyping(const nsACString & aString, int32_t *_retval);
 
   purpleConvIM();
 
 private:
diff --git a/purplexpcom/src/purpleConversation.cpp b/purplexpcom/src/purpleConversation.cpp
--- a/purplexpcom/src/purpleConversation.cpp
+++ b/purplexpcom/src/purpleConversation.cpp
@@ -146,18 +146,31 @@ NS_IMETHODIMP purpleConversation::SetId(
 }
 
 /* void sendMsg (in AUTF8String aMsg); */
 NS_IMETHODIMP purpleConversation::SendMsg(const nsACString& aMsg)
 {
   return NS_ERROR_NOT_IMPLEMENTED;
 }
 
+/* void prepareForSending (in imIOutgoingMessage aMsg,
+                    [optional] out unsigned long aMsgCount,
+                    [array, size_is (aMsgCount), retval] out wstring aMsgs); */
+NS_IMETHODIMP purpleConversation::PrepareForSending(imIOutgoingMessage *aMsg,
+                                                    uint32_t *aMsgCount,
+                                                    char16_t * **aMsgs)
+{
+  *aMsgCount = 0;
+  *aMsgs = nullptr;
+  return NS_OK;
+}
+
 /* void sendTyping (in unsigned long aLength); */
-NS_IMETHODIMP purpleConversation::SendTyping(const nsACString & aString, int32_t *_retval)
+NS_IMETHODIMP purpleConversation::SendTyping(const nsACString & aString,
+                                             int32_t *_retval)
 {
   *_retval = purpleConversation::NO_TYPING_LIMIT;
   return NS_OK;
 }
 
 /* void close (); */
 NS_IMETHODIMP purpleConversation::Close()
 {
